interceptors/Mementifier.cfc

```diff
@@ -210,13 +210,20 @@ component{
 				// Map Items into result object
 				result[ item ] = [];
 				for( var thisIndex = 1; thisIndex <= arrayLen( thisValue ); thisIndex++ ){
-					result[ item ][ thisIndex ] = thisValue[ thisIndex ].getMemento(
-						includes 		= $buildNestedMementoList( includes, item ),
-						excludes 		= $buildNestedMementoList( excludes, item ),
-						mappers 		= mappers,
-						defaults 		= defaults,
-						ignoreDefaults 	= ignoreDefaults
-					);
+					 // only get mementos from relationships that have mementos, in the event that we have an already-serialized array of structs
+					if( isStruct( thisValue[ thisIndex ] ) && structKeyExists( thisValue[ thisIndex ], 'memento' ) ) { // only get mementos from relationships that have mementos
```

Comment:
I think we can change this a bit.  If we just check for the function we need to call, then this will be the only check to do.  Let’s avoid using type checks because they are slow and error prone.

Just change the if statement to this:

```
If( structKeyExists( thisValue[ thisIndex ], “getMemento” )
````

Struct key exists works on objects to verify the function.

Reply:
No problem swapping `memento` for `getMemento`, but without the type check, it can fail if you have an array of simple values because it goes to look at `thisValue[ thisIndex ]` as a structure and it could be a string.